interceptors/tracing/interceptors.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) The go-grpc-middleware Authors.
+// Licensed under the Apache License 2.0.
+
+package tracing
+
+import (
+	"context"
+
+	"google.golang.org/grpc"
+
+	"github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors"
+)
+
+type SpanKind string
+
+const (
+	SpanKindServer SpanKind = "server"
+	SpanKindClient SpanKind = "client"
+)
+
+type reportable struct {
+	tracer Tracer
+}
+
+func (r *reportable) ServerReporter(ctx context.Context, _ interface{}, typ interceptors.GRPCType, service string, method string) (interceptors.Reporter, context.Context) {
```

Comment:
You can use the [`CallMeta`](https://github.com/grpc-ecosystem/go-grpc-middleware/blob/v2/interceptors/reporter.go#L58) struct for embedding server/client related info, and this way you can also reduce redundant code.

Here is an example of an implementation - https://github.com/grpc-ecosystem/go-grpc-middleware/blob/v2/interceptors/logging/interceptors.go#L72